core/include/traccc/fitting/kalman_filter/gain_matrix_smoother.hpp

```diff
@@ -64,7 +64,7 @@ struct gain_matrix_smoother {
         const matrix_type<6, 6>& cur_filtered_cov = cur_filtered.covariance();
 
         // Regularization matrix for numerical stability
-        static constexpr double epsilon = 1e-13;
+        static constexpr scalar epsilon = 1e-13;
```

Comment:
While I'm not the biggest fan of that variable either, wouldn't it be better to use a single variable for this purpose in the project? I.e. `traccc::float_epsilon`?

https://github.com/acts-project/traccc/blob/main/core/include/traccc/definitions/common.hpp#L19

Reply:
I would keep it as it is because their values (1e-5 vs 1e-13) are too different.

Reply:
:+1: